### PR TITLE
CODE RUB: Upgrade Services to Standard 2.10.0 

### PR DESCRIPTION
**Pull request by Pwongchaiya**
Closed: #107

### CODE REVIEW

File: InternTrack.Core.Api/Models/Interns/Exceptions/InternDependecyException.cs [hunk 1]
```diff
@@ -12,5 +13,9 @@ public class InternDependencyException : Xeption
         public InternDependencyException(Xeption innerException)
             : base(message: "Intern dependency error occurred, contact support.", innerException)
         { }
+
+        public InternDependencyException(string message, Exception innerException)
```
Comment (glhays) [inline InternTrack.Core.Api/Models/Interns/Exceptions/InternDependecyException.cs:17]:
```suggestion
        public InternDependencyException(string message, Xeption innerException)
```

File: InternTrack.Core.Api/Models/Interns/Exceptions/InternServiceException.cs [hunk 2]
```diff
@@ -3,14 +3,19 @@
...
 {
     public class InternServiceException : Xeption
     {
-        public InternServiceException(Xeption innterException)
-            : base(message: "Intern service error occured, contact support", innterException)
+        public InternServiceException(Xeption innerException)
+            : base(message: "Intern service error occurred, contact support", innerException)
+        { }
+
+        public InternServiceException(string message,Exception innerException)
```
Comment (glhays) [inline InternTrack.Core.Api/Models/Interns/Exceptions/InternServiceException.cs:17]: same here change Exception to Xeption
and add space after the comma

File: InternTrack.Core.Api/Models/Interns/Exceptions/NotFoundInternException.cs [hunk 3]
```diff
@@ -13,5 +13,9 @@ public class NotFoundInternException : Xeption
         public NotFoundInternException(Guid internId)
             : base(message: $"Couldn't find intern id: {internId}.")
```
Comment (glhays) [inline InternTrack.Core.Api/Models/Interns/Exceptions/NotFoundInternException.cs:14]: If accepted please correct in the test as well.

```suggestion
            : base(message: $"Intern with id: {internId} not found, please correct and try again.")
```

File: InternTrack.Core.Api/Models/Interns/Exceptions/NotFoundInternException.cs [hunk 4]
```diff
@@ -13,5 +13,9 @@ public class NotFoundInternException : Xeption
...
         { }
+
+        public NotFoundInternException(string message,Exception innerException)
```
Comment (glhays) [inline InternTrack.Core.Api/Models/Interns/Exceptions/NotFoundInternException.cs:17]: add space after the comma

File: InternTrack.Core.Api/Models/Interns/Exceptions/FailedInternStorageException.cs [hunk 5]
```diff
@@ -13,5 +13,9 @@ public class FailedInternStorageException : Xeption
         public FailedInternStorageException(Exception innerException)
             : base("Failed intern storage error occurred, contact support.", innerException)
```
Comment (glhays) [inline InternTrack.Core.Api/Models/Interns/Exceptions/FailedInternStorageException.cs:14]: Capitalize intern -> Intern

File: InternTrack.Core.Api/Models/Interns/Exceptions/FailedInternServiceException.cs [hunk 6]
```diff
@@ -11,7 +11,11 @@ namespace InternTrack.Core.Api.Models.Interns.Exceptions
     public class FailedInternServiceException : Xeption
     {
         public FailedInternServiceException(Exception innerException)
-            : base("Failed intern service occured, please contact support", innerException)
+            : base("Failed intern service occurred, please contact support", innerException)
```
Comment (glhays) [inline InternTrack.Core.Api/Models/Interns/Exceptions/FailedInternServiceException.cs:14]:
```suggestion
            : base("Failed Intern service occurred, please contact support", innerException)
```

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.Add.cs [hunk 7]
```diff
@@ -28,10 +28,14 @@ public async Task ShouldThrowCriticalDependencyExceptionOnAddIfSqlErrorOccursAnd
             SqlException sqlException = GetSqlException();
 
             var failedInternStorageException =
-                new FailedInternStorageException(sqlException);
+                new FailedInternStorageException(
+                        "Failed Intern storage error occurred, contact support.", 
+                        sqlException);
```
Comment (hassanhabib) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.Add.cs:33]: Say `message:` and `innerException:` name your parameters

Comment (hassanhabib) [reply InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.Add.cs:33]: Do the same for the rest.

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.Add.cs [hunk 8]
```diff
@@ -28,10 +29,14 @@ public async Task ShouldThrowCriticalDependencyExceptionOnAddIfSqlErrorOccursAnd
             SqlException sqlException = GetSqlException();
 
             var failedInternStorageException =
-                new FailedInternStorageException(sqlException);
+            new FailedInternStorageException(
+                        message: "Failed Intern storage error occurred, contact support.",
```
Comment (glhays) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.Add.cs:33]: Fix indenting here

**Pull request merged**